source/agora/consensus/validation/Block.d

```diff
@@ -112,6 +118,268 @@ public string isInvalidReason (const ref Block block, in ulong prev_height,
     return null;
 }
 
+/*******************************************************************************
+
+    Check the validity of a genesis block
+
+    Follow the Block rules except for the following
+        - Block height is 0
+        - The previous block hash of the block must be empty
+        - The number of transactions in the block is above zero and less than
+          Block.TxsInBlock
+        - All the transactions pass validation, which implies:
+            - Transaction type in block should only be Payment or Freeze
+            - There is empty a input for the transaction
+            - There is one or more outputs of the transaction
+        - All the enrollments pass validation, which implies:
+            - The enrollments refer to freeze tx's in this block
+
+    Params:
+        block = The genesis block to check
```

Comment:
When you change `genesisBlock` to follow the styling convention, you will also have to change this param here (it's currently wrong).